packages/hoppscotch-common/src/composables/pwa.ts

```diff
@@ -11,28 +16,67 @@ export const usePwaPrompt = function () {
     pwaNeedsRefresh,
     (value) => {
       if (value) {
-        toast.show(`${t("app.new_version_found")}`, {
-          duration: 0,
-          action: [
-            {
-              text: `${t("action.dismiss")}`,
-              onClick: (_, toastObject) => {
-                toastObject.goAway(0)
-              },
-            },
-            {
-              text: `${t("app.reload")}`,
-              onClick: (_, toastObject) => {
-                toastObject.goAway(0)
-                refreshAppForPWAUpdate()
-              },
-            },
-          ],
-        })
+        showUpdateToast()
       }
     },
     {
       immediate: true,
     }
   )
+
+  function showUpdateToast() {
+    toast.show(`${t("app.new_version_found")}`, {
+      position: "bottom-left",
+      duration: 0,
+      action: [
+        {
+          text: `${t("action.dismiss")}`,
+          onClick: (_, toastObject) => {
+            toastObject.goAway(0)
+          },
+        },
+        {
+          text: `${t("app.reload")}`,
+          onClick: (_, toastObject) => {
+            toastObject.goAway(0)
+            refreshAppForPWAUpdate()
+          },
+        },
+      ],
+    })
+  }
+}
+
+export const useWhatsNewPrompt = async function () {
+  const persistenceService = getService(PersistenceService)
+
+  const oldVersion = persistenceService.getLocalConfig("hopp_v")
+  const latestMajorVersion = version.split(".").slice(0, 2).join(".")
+
+  if (oldVersion === version) return
+
+  // Skipping the minor version update. e.g. 2024.1.0 -> 2024.1.1
+  // Checking major version update. e.g. 2024.1 -> 2024.2
+
+  const isUpdatedToLatestMajorVersion =
+    oldVersion?.split(".").slice(0, 2).join(".") === latestMajorVersion
+
+  if (isUpdatedToLatestMajorVersion) {
+    const notesUrl = await getReleaseNotes(latestMajorVersion)
+    if (notesUrl) {
+      sonner.custom(markRaw(PWAPrompt), {
+        componentProps: {
+          notesUrl,
+        },
+      })
+    }
+  }
+
+  persistenceService.setLocalConfig("hopp_v", version)
+}
+
+async function getReleaseNotes(v: string) {
+  const res = await fetch(`https://releases.hoppscotch.com/releases/${v}.json`)
+  const { release_notes } = await res.json()
+  return release_notes
 }
```

Comment:
Moving this out into a separate file would be nice because the release notes check is not associated with any PWA stuff.